docs/content/rest-api.md

```diff
@@ -1513,6 +1513,330 @@ The following table summarizes the behavior for partial evaluation results.
+            where a plugin depends on the server being fully initialized before it can fully intialize
+            itself.
+
+#### Status Codes
+- **200** - OPA service is healthy. If the `bundles` option is specified then all configured bundles have
+            been activated. If the `plugins` option is specified then all plugins are in an OK state.
+- **500** - OPA service is not healthy. If the `bundles` option is specified this can mean any of the configured
+            bundles have not yet been activated. If the `plugins` option is specified then at least one
+            plugin is in a non-OK state.
+
+> *Note*: The bundle activation check is only for initial bundle activation. Subsequent downloads
+  will not affect the health check. The [Status](../management-status)
+  API should be used for more fine-grained bundle status monitoring.
+
+#### Example Request
+```http
+GET /health HTTP/1.1
+```
+
+#### Example Request (bundle activation)
+```http
+GET /health?bundles HTTP/1.1
+```
+
+#### Example Request (plugin status)
+```http
+GET /health?plugins HTTP/1.1
+```
+
+#### Example Request (plugin status with exclude)
+```http
+GET /health?plugins&exclude-plugin=decision-logs&exclude-plugin=status HTTP/1.1
+```
+
+#### Healthy Response
+```http
+HTTP/1.1 200 OK
+Content-Type: application/json
+```
+```json
+{}
+```
+
+#### Unhealthy Response
+```http
+HTTP/1.1 500 Internal Server Error
+Content-Type: application/json
+```
+```json
+{
+  "error": "not all plugins in OK state"
+}
+```
+
+Other error messages include:
+
+- `"unable to perform evaluation"`
+- `"not all configured bundles have been activated"`
+
+### Custom Health Checks
+
+The Health API includes support for "all or nothing" checks that verify
+configured bundles have activated and plugins are operational. In some cases,
+health checks may need to perform fine-grained checks on plugin state or other
+internal components. To support these cases, use the policy-based Health API.
+
+By convention, the `/health/live` and `/health/ready` API endpoints allow you to
+use Rego to evaluate the current state of the server and its plugins to
+determine "liveness" (when OPA is capable of receiving traffic) and "readiness"
+(when OPA is ready to receive traffic). Policy for the `live` and `ready` rules
+is defined under package `system.health`.
+
+> The "liveness" and "readiness" check convention comes from
+> [Kubernetes](https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/)
+> but they are just conventions. You can implement your own check endpoints
+> under the `system.health` package as needed. Any rules implemented inside of
+> `system.health` will be exposed at `/health/<rule-name>`.
+
+#### Policy Examples
+
+Here is a basic health policy for liveness and readiness. In this example, OPA is live once it is
+able to process the `live` rule. OPA is ready once all plugins have entered the OK state at least once.
+
+```rego
```

Comment:
We should use some `live:...` variant here to get highlighting on the website. Perhaps `live:policy_example:module:read_only`